test/unit_tests/braket/default_simulator/cirq/test_openqasm_conversion.py

```diff
@@ -0,0 +1,216 @@
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[1] b;
+        qubit[1] q;
+        rz(0.2) q[0];
+        b[0] = measure q[0];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[2] b;
+        qubit[2] q;
+        x q[0];
+        x q[1];
+        b[0] = measure q[0];
+        b[1] = measure q[1];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[3] b;
+        qubit[3] q;
+        y q[0];
+        y q[1];
+        y q[2];
+        b[0] = measure q[0];
+        b[1] = measure q[1];
+        b[2] = measure q[2];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[1] b;
+        qubit[1] q;
+        s q[0];
+        b[0] = measure q[0];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[1] b;
+        qubit[1] q;
+        t q[0];
+        b[0] = measure q[0];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[2] b;
+        qubit[2] q;
+        cphaseshift(0.15) q[1], q[0];
+        b[0] = measure q[0];
+        b[1] = measure q[1];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[2] b;
+        qubit[2] q;
+        swap q[0], q[1];
+        b[0] = measure q[0];
+        b[1] = measure q[1];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[3] b;
+        qubit[3] q;
+        ccnot q[0], q[1], q[2];
+        b[0] = measure q[0];
+        b[1] = measure q[1];
+        b[2] = measure q[2];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[3] b;
+        qubit[3] q;
+        cswap q[0], q[1], q[2];
+        b[0] = measure q[0];
+        b[1] = measure q[1];
+        b[2] = measure q[2];
+        """
+        ),
+        (
+            """
+        OPENQASM 3.0;
+        bit[12] b;
+        qubit[12] q;
+        rx(0.15) q[0];
+        ry(0.2) q[1];
+        rz(0.25) q[2];
+        h q[3];
+        cnot q[0], q[2];
+        cnot q[1], q[3];
+        x q[1];
+        x q[3];
+        y q[1];
+        y q[3];
+        z q[1];
+        z q[2];
+        s q[2];
+        t q[1];
+        cphaseshift(0.15) q[4], q[2];
+        swap q[4], q[5];
+        ccnot q[6], q[7], q[8];
+        cswap q[9], q[10], q[11];
+        b[0] = measure q[0];
+        b[1] = measure q[1];
+        b[2] = measure q[2];
+        b[3] = measure q[3];
+        b[4] = measure q[4];
+        b[5] = measure q[5];
+        b[6] = measure q[6];
+        b[7] = measure q[7];
+        b[8] = measure q[8];
+        b[9] = measure q[9];
+        b[10] = measure q[10];
+        b[11] = measure q[11];
+        """
+        ),
+    ),
+)
+def test_openqasm_conversion_to_cirq(openqasm):
+    circuit = Interpreter(ProgramContext()).build_circuit(
+        source=openqasm,
+    )
+
+    assert isinstance(circuit, cirq.Circuit)
```

Comment:
can we add a correctness check here? Maybe we can compare the state vectors of the two circuits?

Reply:
That would mean I have to run it on a simulator right?

I was thinking to do it and then if I would directly access a cirq simulator here. But i was thinking to create a plugin like below in the end once I complete the implementation of all the available gates.
I have to create a PR to add support for localSimulator("cirq") in the sdk repo. 

Do you think it is better to do it now by directly accessing the cirq simulator  and refactor it later to use the simulator via the plugin.

Reply:
`LocalSimulator("cirq")` is an entirely separate consideration, since it lives in the BDK and we won't be referencing BDK code at all in this repo. Take a look at [this test](https://github.com/aws/amazon-braket-default-simulator-python/blob/main/test/unit_tests/braket/default_simulator/openqasm/test_interpreter.py#L1006) for an example of how to simulate a circuit a local sim unit test. For cirq, you can extract the state vector using its library methods, maybe something like [this](https://quantumai.google/reference/python/cirq/final_state_vector)

Reply:
Thank you.

I just have a small query about how would I know what the expected vector is?

I have generated the openQasm strings from the Braket circuits.

Do I have do something like this
`simulation = StateVectorSimulation(2, 1, 1)
    simulation.evolve(circuit.instructions)`
  to obtain a expected value ?
   Or is there a better way and more accurate way of doing it
   
   
For example:
` 
circ = BCircuit.Circuit().cnot(control=0, target=1, power =0.5)
circuit = Interpreter(ProgramContext()).build_circuit(
    source=circ.to_ir(IRType.OPENQASM).source,
)
 cirq.final_state_vector(circuit)
 `
  
  Output: 
`[1.+0.j 0.+0.j 0.+0.j 0.+0.j]`  
 
In a nut shell my query is what is the best way to know the expected value and compare it to the above value since I dont know how to make the calculations manually.

Reply:
I was thinking of it but I was not sure. Thanks for the references!

Reply:
This caught a bug.

ctrl_modifiers = [bit^1 for bit in ctrl_modifiers]

Along with the added check,  fixed the bug. Nice catch : )